source/Core/Src/settingsGUI.cpp

```diff
@@ -595,49 +595,17 @@ static bool setResetSettings(void) {
 
 static void displayResetSettings(void) {}
 
-static void setTipOffset() {
-  uint16_t setoffset = 0;
-
-  // If the thermo-couple at the end of the tip, and the handle are at
-  // equilibrium, then the output should be zero, as there is no temperature
-  // differential.
-  while (setoffset == 0) {
-    uint32_t offset = 0;
-    for (uint8_t i = 0; i < 16; i++) {
-      offset += getTipRawTemp(1);
-      // cycle through the filter a fair bit to ensure we're stable.
-      OLED::clearScreen();
-      OLED::setCursor(0, 0);
-      OLED::print(SymbolDot, FontStyle::LARGE);
-      for (uint8_t x = 0; x < (i / 4); x++)
-        OLED::print(SymbolDot, FontStyle::LARGE);
-      OLED::refresh();
-      osDelay(100);
-    }
-    setoffset = TipThermoModel::convertTipRawADCTouV(offset / 16, true);
-  }
-  setSettingValue(SettingsOptions::CalibrationOffset, setoffset);
-  OLED::clearScreen();
-  OLED::setCursor(0, 0);
-  OLED::drawCheckbox(true);
-  OLED::printNumber(setoffset, 4, FontStyle::LARGE);
-  OLED::refresh();
-  osDelay(1200);
-}
-
-// Provide the user the option to tune their own tip if custom is selected
-// If not only do single point tuning as per usual
 static bool setCalibrate(void) {
 
   if (userConfirmation(translatedString(Tr->SettingsCalibrationWarning))) {
     // User confirmed
-    // So we now perform the actual calculation
-    setTipOffset();
+    // So we now set the tick
+    setSettingValue(SettingsOptions::CalibrateCJC, 1);
   }
   return false;
 }
 
-static void displayCalibrate(void) {}
+static void displayCalibrate(void) { OLED::drawCheckbox(getSettingValue(SettingsOptions::CalibrateCJC)); }
```

Comment:
@Ralim
I don't know what I destroyed here, but this is not working as anticipated! 😅

Reply:
The draw Checkbox should be right. I would compare against `displayAdvancedSolderingScreens` & the advanced soldering toggle to check against.

Reply:
Yes, the drawCheckbox works flawlessly,  but my attempt to make the **`SettingsCalibrationWarning`** to be displayed before the checkbox gets set, does not work!

Actually I was referring to this part of code:
```C#
static bool setCalibrate(void) {

  if (userConfirmation(translatedString(Tr->SettingsCalibrationWarning))) {
    // User confirmed
    // So we now perform the actual calculation
    setTipOffset();
    // So we now set the tick
    setSettingValue(SettingsOptions::CalibrateCJC, 1);
  }
  return false;
}
```

Reply:
I don't understand why this works:
```C#
static bool setResetSettings(void) {
  if (userConfirmation(translatedString(Tr->SettingsResetWarning))) {
    resetSettings();
    warnUser(translatedString(Tr->SettingsResetMessage), 10 * TICKS_SECOND);
  }
  return false;
}

static void displayResetSettings(void) {}
```

... but not this:
```C#
static bool setCalibrate(void) {
  if (userConfirmation(translatedString(Tr->SettingsCalibrationWarning))) {
    // User confirmed
    // So we now set the tick
    setSettingValue(SettingsOptions::CalibrateCJC, 1);
  }
  return false;
}

static void displayCalibrate(void) { OLED::drawCheckbox(getSettingValue(SettingsOptions::CalibrateCJC)); }
```

Reply:
Honestly, I think that _should_ work.
Maybe double check git hash in debug menu to make sure it wasnt a bad flash?
What do you see?

Reply:
1. I tried it several times and I am getting always the same result.
2. The Calibrate entry appears with an empty checkbox and after pressing A it just gets checked without the warning message.

Reply:
Ah figured it out.
You need to change the `SettingsOptions::CalibrateCJC` to `SettingsOptions::SettingsOptionsLength` to make it not use the automatic settings toggle and instead call your setSettingValue() function

Reply:
You can still use `setSettingValue(SettingsOptions::CalibrateCJC, 0);` / `setSettingValue(SettingsOptions::CalibrateCJC, 1);` Normally.

Just to cut down on code side, if that value is set, it assumes it cant just tell the settings code to increment the value. Which cuts down on code duplication (and space) since most values just increment and wrap.

Reply:
1. Will try that after I have resolved the conflicts after the latest merges.
2. I don't quite get what exactly is meant by that, will have to read it a few more times.

Reply:
So you still use `setSettingValue` to set the value to 0/1 for on/off as per normal.
But in the menu definition you need to swap out the value.
To get past this check https://github.com/Ralim/IronOS/blob/c99a1995202bb5b173925ad9eaa8e6a247f984aa/source/Core/Src/settingsGUI.cpp#L957

Reply:
So this finally works as I intended it to do!

If the tick is not set, the **`warning message`** is displayed if **`A`** gets pressed.
After confirming with **`A`** the tick is set. If **`B`** is used, though, it will not  be set!

However, if the tick is already set, another press on **`A`** will remove the tick again.